hcl2template/types.source.go

```diff
@@ -57,12 +57,19 @@ func (p *Parser) startBuilder(source *SourceBlock, ectx *hcl.EvalContext) (packe
 		return nil, diags, nil
 	}
 
-	generatedVars, warning, err := builder.Prepare(decoded)
+	generatedVars, warning, err := builder.Prepare(source.builderVariables(), decoded)
```

Comment:
Here we use the properties of mapstructure to first pass on the builder variables then set the decoded values from an hcl config file. This allows to not have to manually default this for every builder/provisioner/post-processor